### PR TITLE
Omit image tags in gitops targets in k8s_deploy bazel rule

### PR DESCRIPTION
**Pull request by arturo-skydio**
- This is so we can still use tags in `.show` and `.apply` to deploy from bazel commands but be able to rely consistently on bazel gitops targets remaining the same and then we can be sure to test gitops files in CI.
- The design of `rules_gitops` was that the gitops file commited would be the source of truth with what is deployed, but at Skydio we deploy with argocd and spinnaker outside of the git loop

Tests:
Tested with `skyops` and works as it should

Topic: omit-image-in-gitops
Reviewers: daniel-benton

### CODE REVIEW

**Comment by arturo-skydio** [comment]
Reviews in this chain:
└**https://github.com/Skydio/rules_gitops/pull/8** **Omit image tags in gitops targets in k8s_deploy bazel rule**

**Comment by arturo-skydio** [comment]
| # | head | base | diff | date | summary |
| - | - | - | - | - | - |
| 0 | [50161e82](/Skydio/rules_gitops/commit/50161e82eb27385dff8167781fcc2df31d311b0c) | [1ae7367b](/Skydio/rules_gitops/commit/1ae7367b05941b2ed7c27cecd5b98b8023100574) | [diff](/Skydio/rules_gitops/compare/1ae7367b05941b2ed7c27cecd5b98b8023100574..50161e82eb27385dff8167781fcc2df31d311b0c) | Dec 9 16:43 PM |  2 files changed, 34 insertions(+), 2 deletions(-) |
| 1 | [0ef261a9](/Skydio/rules_gitops/commit/0ef261a99a619ed81a1b522024dc03888d5ca732) | [1ae7367b](/Skydio/rules_gitops/commit/1ae7367b05941b2ed7c27cecd5b98b8023100574) | [diff](/Skydio/rules_gitops/compare/50161e82eb27385dff8167781fcc2df31d311b0c..0ef261a99a619ed81a1b522024dc03888d5ca732) | Dec 9 16:45 PM |  1 file changed, 1 deletion(-) |
| 2 | [c8fd6b4a](/Skydio/rules_gitops/commit/c8fd6b4ae3eeb5fa63cf5cb7b8fb06f0e4177951) | [1ae7367b](/Skydio/rules_gitops/commit/1ae7367b05941b2ed7c27cecd5b98b8023100574) | [diff](/Skydio/rules_gitops/compare/0ef261a99a619ed81a1b522024dc03888d5ca732..c8fd6b4ae3eeb5fa63cf5cb7b8fb06f0e4177951) | Dec 9 17:05 PM |  1 file changed, 6 insertions(+), 2 deletions(-) |

File: skylib/k8s.bzl [hunk 1]
```diff
@@ -265,6 +266,32 @@ def k8s_deploy(
             image_tag_patches = image_tag_patches,
             tags = tags,
         )
+        kustomize(
+            name = name + "_gitops",
```
Comment (arturo-skydio) [inline skylib/k8s.bzl:270]: created a duplicate of the above kustomize function call, but create another target that will be used exclusively for gitops

**Review by daniel-benton-skydio — APPROVED** [review]
Nice!  I think this change makes sense to me... Any value in contributing this back to `adobe/rules_gitops`?

As a follow-up: is there a way to decouple the manifests from the image itself?  i.e. if we run `.gitops`, can we avoid building images?  Seems like if we're deliberately skipping including the image hash, we might want to avoid building the image altogether.

**Comment by arturo-skydio** [comment]
> Nice! I think this change makes sense to me... Any value in contributing this back to `adobe/rules_gitops`?
> 
> As a follow-up: is there a way to decouple the manifests from the image itself? i.e. if we run `.gitops`, can we avoid building images? Seems like if we're deliberately skipping including the image hash, we might want to avoid building the image altogether.

Contributing this might make sense, I'll put up a PR, although they might not be interested in adding that logic since it really negates a bit portion of their workflow.

I think we could skip running the image push step, although its really tightly coupled to the current functions, might have to do some deeper refactoring which is why I left that change out of this initial PR

File: skylib/kustomize/kustomize.bzl [hunk 2]
```diff
@@ -405,6 +409,10 @@ def _remove_prefixes(s, prefixes):
         s = _remove_prefix(s, prefix)
     return s
 
+def _remove_gitops_suffix(s):
+    s = s.replace("_gitops.yaml", ".yaml")
```
Comment (arturo-skydio) [inline skylib/kustomize/kustomize.bzl:413]: this replacement string patterns are so they don't rewrite all of our gitops targets, there might be a better way to do this, but this is the best I could come up with for now

**Pull request merged**